components/action-mgt/org.wso2.carbon.identity.action.management/src/main/java/org/wso2/carbon/identity/action/management/constant/ActionMgtSQLConstants.java

```diff
@@ -0,0 +1,82 @@
+ */
+public class ActionMgtSQLConstants {
+
+    private ActionMgtSQLConstants() {
+
+    }
+
+    /**
+     * Column Names.
+     */
+    public static class Column {
+
+        public static final String ACTION_UUID = "UUID";
+        public static final String ACTION_TYPE = "TYPE";
+        public static final String ACTION_NAME = "NAME";
+        public static final String ACTION_DESCRIPTION = "DESCRIPTION";
+        public static final String ACTION_STATUS = "STATUS";
+        public static final String ACTION_COUNT = "COUNT";
+        public static final String ACTION_ENDPOINT_UUID = "ACTION_UUID";
+        public static final String ACTION_ENDPOINT_PROPERTY_NAME = "PROPERTY_NAME";
+        public static final String ACTION_ENDPOINT_PROPERTY_VALUE = "PROPERTY_VALUE";
+        public static final String TENANT_ID = "TENANT_ID";
+
+        private Column() {
+
+        }
+    }
+
+    /**
+     * Queries.
+     */
+    public static class Query {
+
+        public static final String ADD_ACTION_TO_ACTION_TYPE = "INSERT INTO IDN_ACTION (UUID, TYPE, NAME, " +
+                "DESCRIPTION, STATUS, TENANT_ID) VALUES (:UUID;, :TYPE;, :NAME;, :DESCRIPTION;, :STATUS;, :TENANT_ID;)";
+        public static final String ADD_ACTION_ENDPOINT_PROPERTIES = "INSERT INTO IDN_ACTION_ENDPOINT (ACTION_UUID, " +
+                "PROPERTY_NAME, PROPERTY_VALUE, TENANT_ID) VALUES (:ACTION_UUID;, :PROPERTY_NAME;, :PROPERTY_VALUE;, " +
+                ":TENANT_ID;)";
+        public static final String GET_ACTION_BASIC_INFO_BY_ID = "SELECT TYPE, NAME, DESCRIPTION, STATUS FROM " +
+                "IDN_ACTION WHERE UUID = :UUID; AND TENANT_ID = :TENANT_ID;";
+        public static final String GET_ACTION_ENDPOINT_INFO_BY_ID = "SELECT PROPERTY_NAME, PROPERTY_VALUE FROM " +
+                "IDN_ACTION_ENDPOINT WHERE ACTION_UUID = :ACTION_UUID; AND TENANT_ID = :TENANT_ID;";
+        public static final String GET_ACTIONS_BASIC_INFO_BY_ACTION_TYPE = "SELECT UUID, TYPE, NAME, DESCRIPTION," +
+                " STATUS FROM IDN_ACTION WHERE TYPE = :TYPE; AND TENANT_ID = :TENANT_ID;";
+        public static final String UPDATE_ACTION_BASIC_INFO = "UPDATE IDN_ACTION SET NAME = :NAME;, DESCRIPTION = " +
+                ":DESCRIPTION; WHERE UUID = :UUID; AND TYPE = :TYPE; AND TENANT_ID = :TENANT_ID;";
+        public static final String DELETE_ACTION_ENDPOINT_PROPERTIES = "DELETE FROM IDN_ACTION_ENDPOINT WHERE " +
+                "ACTION_UUID = :ACTION_UUID; AND TENANT_ID = :TENANT_ID;";
+        public static final String DELETE_ACTION = "DELETE FROM IDN_ACTION WHERE UUID = :UUID; AND TYPE = :TYPE;" +
+                " AND TENANT_ID = :TENANT_ID;";
+        public static final String CHANGE_ACTION_STATUS = "UPDATE IDN_ACTION SET STATUS = :STATUS; WHERE UUID = " +
+                ":UUID; AND TYPE = :TYPE; AND TENANT_ID = :TENANT_ID;";
+        public static final String GET_ACTIONS_COUNT_PER_ACTION_TYPE = "SELECT TYPE, COUNT(TYPE) AS COUNT" +
```

Comment:
Shouldn't we count ACTION_UUID. If there's no benefit  given it's the primary key this is fine

Reply:
Fixed with https://github.com/wso2/carbon-identity-framework/pull/5771/commits/a96c6e8c2f905af6c7fd842a69a4d972997473a1